nixos/lib/make-ext4-fs.nix

```diff
@@ -37,32 +37,27 @@ pkgs.stdenv.mkDerivation {
       ${populateImageCommands}
       )
 
-      # Add the closures of the top-level store objects.
-      storePaths=$(cat ${sdClosureInfo}/store-paths)
+      echo "Preparing store paths for image..."
+
+      # Create nix/store before copying path
+      mkdir -p ./rootImage/nix/store
+
+      xargs -I % cp -a --reflink=auto % -t ./rootImage/nix/store/ < ${sdClosureInfo}/store-paths
+      cp -a --reflink=auto ./files/* -t ./rootImage/
```

Comment:
This would leave hidden files behind if added using `populateImageCommands`.

Looks like we can use bash features to get this working. The following could be used with a subshell:

```
(
  GLOBIGNORE=".:.."
  shopt -u dotglob
  cp -a --reflink=auto ./files/* -t ./rootImage/
)
```

Reply:
Done!